### PR TITLE
Fix build warnings & errors when building non-US versions

### PR DESCRIPTION
**Pull request by Arceveti**

### CODE REVIEW

**Review by gheskett — APPROVED** [review]
Moving boot issues into 2.1; this can be merged into 2.0.2 IMO. That said, maybe it's better to also move this to 2.1 so users don't waste their time trying to use EU on 2.0.2?

File: src/audio/effects.c [hunk 1]
```diff
@@ -370,12 +370,12 @@ s32 adsr_update(struct AdsrState *adsr) {
             adsr->currentHiRes = (adsr->current << 0x10);
 #endif
             adsr->state = ADSR_STATE_LOOP;
+            // fallthrough
 
+        case ADSR_STATE_LOOP:
 #ifdef VERSION_SH
             restart:
 #endif
```
Comment (thecozies) [inline src/audio/effects.c:378]: this seems to change shindou functionality 🤔

File: src/audio/heap.c [hunk 2]
```diff
@@ -421,6 +421,7 @@ void *alloc_bank_or_seq(struct SoundMultiPool *arg0, s32 arg1, s32 size, s32 arg
             break;
 
         case 2:
+        default:
```
Comment (thecozies) [inline src/audio/heap.c:424]: you're sure this should be a default?

File: src/audio/heap.c [hunk 3]
```diff
@@ -770,6 +771,7 @@ void *get_bank_or_seq_inner(s32 poolIdx, s32 arg1, s32 bankId) {
             loadedPool = &gBankLoadedPool;
             break;
         case 2:
+        default:
```
Comment (thecozies) [inline src/audio/heap.c:774]: this as well?

**Review by thecozies — CHANGES_REQUESTED** [review]
couple questions

File: src/audio/heap.c (hunk 2 above)
Comment (gheskett) [reply src/audio/heap.c:424]: Assuming the error here is arg0 being uninitialized, I would just set it to NULL at the beginning of the function and remove the default.

File: src/audio/heap.c (hunk 3 above)
Comment (gheskett) [reply src/audio/heap.c:774]: Yes, set loadedPool to NULL at declaration and remove the default case.

File: src/audio/effects.c (hunk 1 above)
Comment (gheskett) [reply src/audio/effects.c:378]: Yeah, this would be a pretty noticeable bug. Please fix.

File: src/game/ingame_menu.c [hunk 4]
```diff
@@ -1369,6 +1367,7 @@ void print_peach_letter_message(void) {
     void **dialogTable;
     gInGameLanguage = eu_get_language();
     switch (gInGameLanguage) {
+        default:
```
Comment (gheskett) [inline src/game/ingame_menu.c:1370]: This is valid, but bad practice to put `default` somewhere other than the end of a switch case.

File: src/menu/star_select.c [hunk 5]
```diff
@@ -313,6 +313,7 @@ void print_act_selector_strings(void) {
 
 #if MULTILANG
     switch (language) {
+        default:
```
Comment (gheskett) [inline src/menu/star_select.c:316]: Same thing here; valid but discouraged

**Review by gheskett — CHANGES_REQUESTED** [review]
At the very minimum, the SH bug should be fixed. Everything else is recommended but won't really hurt functionally.

File: src/audio/heap.c (hunk 2 above)
Comment (Arceveti) [reply src/audio/heap.c]: done

File: src/audio/heap.c (hunk 3 above)
Comment (Arceveti) [reply src/audio/heap.c]: done

File: src/audio/effects.c (hunk 1 above)
Comment (Arceveti) [reply src/audio/effects.c]: I reverted it and added another `// fallthrough` instead to fix the warning.

File: levels/intro/geo.c [hunk 6]
```diff
@@ -72,11 +72,11 @@ const GeoLayout intro_geo_mario_head_regular[] = {
...
-#if defined(VERSION_SH)
-   GEO_ZBUFFER(0),
-   GEO_OPEN_NODE(),
-      GEO_ASM(INTRO_CONTEXT_NORMAL, geo_intro_rumble_pak_graphic),
-   GEO_CLOSE_NODE(),
+#if defined(ENABLE_RUMBLE)
+      GEO_ZBUFFER(0),
+      GEO_OPEN_NODE(),
+         GEO_ASM(INTRO_CONTEXT_NORMAL, geo_intro_rumble_pak_graphic),
+      GEO_CLOSE_NODE(),
```
Comment (gheskett) [inline levels/intro/geo.c:79]: Why the tabs here?

**Review by gheskett — APPROVED** [review]
Not everything has been addressed, but I would still be okay with merging at this stage.

File: levels/intro/geo.c (hunk 6 above)
Comment (gheskett) [reply levels/intro/geo.c:79]: Nevemind, disregard

**Pull request merged**